Applications/Spire/Source/Ui/CalendarDatePicker.cpp

```diff
@@ -46,6 +50,82 @@ namespace {
   }
 }
 
+class CalendarModel {
+  public:
+    static const int DAY_COUNT = 42;
+
+    using Dates = std::array<date, DAY_COUNT>;
+
+    explicit CalendarModel(date current) {
+      set_current(m_current);
+    }
+
+    void set_current(date current) {
+      if(current != m_current) {
+        m_current = current;
+        auto day = date(m_current.year(), m_current.month(), 1);
+        if(day.day_of_week() != 0) {
+          day += days(-day.day_of_week());
+        }
+        for(auto i = 0; i < DAY_COUNT; ++i) {
+          m_dates[i] = day;
+          day += days(1);
+        }
+      }
+    }
+
+    const Dates& get_dates() const {
+      return m_dates;
+    }
+
+  private:
+    Dates m_dates;
+    date m_current;
+};
+
+class CalendarListModel : public ListModel {
+  public:
+    CalendarListModel(std::shared_ptr<DateModel> model)
+        : m_model(std::move(model)),
+          m_calendar_model(m_model->get_current()),
+          m_current_connection(m_model->connect_current_signal(
+            [=] (auto current) { on_current(current); })) {
+      on_current(m_model->get_current());
+    }
+
+    const std::shared_ptr<DateModel>& get_model() const {
+      return m_model;
+    }
+
+    int get_size() const override {
+      return m_dates.size();
+    }
+
+    const std::any& at(int index) const override {
+      return m_dates[index];
+    }
+
+    connection connect_operation_signal(
+        const OperationSignal::slot_type& slot) const override {
+      return m_transaction.connect_operation_signal(slot);
+    }
+
+  private:
+    std::shared_ptr<DateModel> m_model;
+    scoped_connection m_current_connection;
+    CalendarModel m_calendar_model;
+    std::array<std::any, CalendarModel::DAY_COUNT> m_dates;
+    ListModelTransactionLog m_transaction;
```

Comment:
Why is this transaction log used?

Reply:
Okay that's good then.

Reply:
Well, now I'm confused. As a private implementation of the CalendarDatePicker I know an update in the CalendarListModel won't result in additional transactions that cause reentrancy issues, but do I address reentrancy here or trust that the CalendarDatePicker won't abuse its own models?

I'm leaning back towards keeping the transaction log on the chance that the CalendarDatePicker is updated and these assumptions on safety break.

Reply:
I believe the update is considered a transaction.

Reply:
Excellent, that makes a lot more sense now. You want a transaction to update all of the dates in one shot without interruption. Look over the TransactionLog's API and unit tests to get a sense of how it works and if you have questions, ask them on Asana.